lib/fdo.c

```diff
@@ -1736,3 +1738,35 @@ static bool _STATE_Shutdown_Error(void)
 	/* Return false becuase there has been a failure. */
 	return false;
 }
+
+#ifdef WIN32
+#include <stdarg.h>
+int snprintf_s_i(char *buffer, size_t count, const char *format, ...)
+{
+	va_list args;
+	int retval;
+
+    va_start(args, format);
+	retval = vsnprintf(buffer, count, format, args);
+	va_end(args);
+
+	return retval;
+
+
+}
+
+
+int snprintf_s_si(char *buffer, size_t count, const char *format, const char* encoding, ...)
+{
+	va_list args;
```

Comment:
If it is same as teh previous snprintf_s_i, why not invoke it directly from here ?